src/position.cc

```diff
@@ -595,3 +595,16 @@ void Position::print() const
         std::cout << std::endl;
     }
 }
+
+bool Position::insufficient_material() const
+{
+	if(bb::count(occ()) > 3)
+		return false;
+	if(bb::count(occ(KNIGHT)) == 1)
+		return true;
+	if(bb::count(occ(BISHOP)) == 1)
+		return true;
+	if(bb::count(occ()) == 2)
+		return true;
+	return false;
+}
```

Comment:
Can you write this in a cleaner way ?

One line should suffice, and you don't need any `if` and `count()`.